### ISSUE TITLE
Implement flake8-error-message practices

### ISSUE THREAD

**Issue by nstarman** [astropy/astropy#14793]
### What is the problem this feature will solve?

In errors of the form
```
raise Exception(f"some message: {more_info}")
```

The error traceback includes the construction of the Exception message, particularly the construction of the f-string. This adds one-to-two extraneous levels to every error traceback 😲.
One simple change -- pre-assigning the string -- can help reduce the traceback, making it shorter and easier to parse to find the real problem.

```
msg = f"some message: {more_info}"
raise Exception(msg)
```

Now the construction of the message is not included in the traceback. Begone f-string!

### Describe the desired outcome

Astropy turns on, package by package using our new Ruff config from #14744, the 3 settings in https://beta.ruff.rs/docs/rules/#flake8-errmsg-em. These are all **autofix** so will be automatically done for all commits using pre-commit or trivially fixed via the PR comment "pre-commit.ci autofix" for those without local CI tests.
TLDR: no human intervention necessary; same-to-nicer tracebacks guaranteed.

### Additional context

@mhvk, from discussion in #14780

**Comment by nstarman** [comment]
> ``raise ... from exc``

👍 to that.

**Comment by bjbomfim** [comment]
Hello!
I was wondering if I could work in this issue? Or somebody else is working on it?

**Comment by pllim** [comment]
I strongly object this being applied across the repo. If you are adding new code and you want to do it this way, that is your call. But a lot of maintainers are fatigued by all the style refactoring at this point, so I am closing this issue. Thank you.
